core/common/src/ContourDisplay.ts

```diff
@@ -0,0 +1,387 @@
+  /** The pattern for a major or minor contour line. Defaults to [[LinePixels.Solid]]. */
+  public readonly pattern: LinePixels;
+
+  /** Returns true if `this` and `other` are logically equivalent. */
+  public equals(other: ContourStyle): boolean {
+    if (!this.color.equals(other.color) || this.pixelWidth !== other.pixelWidth || this.pattern !== other.pattern) {
+      return false;
+    }
+    return true;
+  }
+
+  /** Performs ordered comparison of two contour styles.
+   * @param lhs First contour style to compare
+   * @param rhs Second contour style to compare
+   * @returns 0 if lhs is equivalent to rhs, a negative number if lhs compares less than rhs, or a positive number if lhs compares greater than rhs.
+   * @public
+   */
+  public static compare(lhs: ContourStyle, rhs: ContourStyle): number {
+    let diff = 0;
+    if ((diff = lhs.color.compareTo(rhs.color)) !== 0)
+      return diff;
+    if ((diff = compareNumbers(lhs.pixelWidth, rhs.pixelWidth)) !== 0)
+      return diff;
+    if ((diff = compareNumbers(lhs.pattern, rhs.pattern)) !== 0)
+      return diff;
+
+    return diff;
+  }
+
+  private constructor(json?: ContourStyleProps) {
+    if (undefined === json) {
+      this.color = RgbColor.fromJSON({r: 0, g: 0, b: 0});
+      this.pixelWidth = 1;
+      this.pattern = 0;
+    } else {
+      this.color = json.color ? RgbColor.fromJSON(json.color) : RgbColor.fromJSON({r: 0, g: 0, b: 0});
+      this.pixelWidth = json.pixelWidth ?? 1;
+      this.pattern = json.pattern ?? 0;
+    }
+  }
+
+  public static fromJSON(json?: ContourStyleProps) {
+    return json ? new ContourStyle(json) : new ContourStyle({});
+  }
+
+  public toJSON(): ContourStyleProps {
+    const props: ContourStyleProps = {};
+
+    if (!this.color.equals(RgbColor.fromJSON({r: 0, g: 0, b: 0})))
+      props.color = this.color.toJSON();
+
+    if (1 !== this.pixelWidth)
+      props.pixelWidth = this.pixelWidth;
+
+    if (0 !== this.pattern)
+      props.pattern = this.pattern;
+
+    return props;
+  }
+}
+
+/** JSON representation of a [[Contour]].
+ * @public
+ */
+export interface ContourProps {
+  /** See [[Contour.majorStyle]]. */
+  majorStyle?: ContourStyleProps;
+  /** See [[Contour.minor]]. */
+  minorStyle?: ContourStyleProps;
+  /** See [[Contour.minorInterval]]. */
+  minorInterval?: number;
+  /** See [[Contour.majorIntervalCount]]. */
+  majorIntervalCount?: number;
+  /** See [[Contour.showGeometry]] */
+  showGeometry?: boolean;
+}
+
+/** Describes how to generate and style contour lines for geometry within a single [[ContourGroup]].
+ * Contours provide a way to visualize elevation within a 3d scene by drawing lines at fixed intervals along the z-axis.
+ * There are actually 2 kinds of contour lines: major and minor. Each kind can be styled independently.
+ * A contour line is generated every [[minorInterval]] meters. Every `nth` line will be a *major* contour, where `n` = [[majorInterval]]; the intervening lines will
```

Comment:
Document what happens if `majorInterval` is less than 1 or not an integer, and what happens if `minorInterval` is <= 0.

Reply:
@DStradley please take a look when you can.

Reply:
Major interval is rounded to an integer.  If it is 1, no minor intervals will be drawn.  If it is <= 0, no major intervals will be drawn.  (The latter is observed behavior, probably due to nan's in the shader.)
Minor intervals = 0 causes no contour lines to be drawn (likely due to shader nan).
Minor intervals < 0 behave as if they were positive (abs(interval, and not explicitly checked for).
Should we add some additional checks for Major <=0 and minor <=0, and if so what should be the behavior?

Reply:
The docs should say what the allowed values are and what happens to values that don't meet that criterion (e.g., round non-integers, clamp to allowed range, whatever). Ideally don't rely on NaN in shader just happening to produce a result.

Reply:
I pushed code changes to keep in range, and @markschlosseratbentley pushed documentation changes for this, so should be done.